### DISCUSSION TITLE
How to use player.pause() before PlayerBar in/with container

### DISCUSSION THREAD

**Discussion by jeejeestudio** [physia/kflutter#20] [Q&A] [answered]
Hi Mohamad,

I've updated to 0.2.4.

First I thought palyers was a typo but it seems not:
I'm using this for each player:
```
Container( //by using theme dark in Main, no color background is needed
                           width: 400.0,
                           height: 50,
                           child: PlayerBar(player: player1, options: [])),
```
How do I use `PlayerController.palyers.forEach((_player)=>_player.pause());`
or `player.others.forEach((player) => player.pause());` in/with the above container?

I tried both before container and in child just before the playerbar, of course i've changed the ; for , else the rest is not working anymore. For some of course I needed to use the correct player name like player1
I'm getting this error:
_This expression has a type of 'void' so its value can't be used._

You see i'm a novice, but learning. Also looking for possible solutions concerning error messages on Stack Overflow.

Thanks!

Regards,
Jerry

**Comment by mohamadlounnas** [comment, marked as answer]
hi @jeejeestudio 

add a stream listener for the player,
example:
```dart
var player = Player.asset("/assets/sound.mp3");
var subscription = player.streams.playing.stream.listen((bool playing) {
   if (playing) {
       player.others.forEach((player) => player.pause());
   }
});
```
il u have a list of players, do it for every player, and dont forget to close the subscription on dispose method.

**Reply by jeejeestudio** [reply]
Hi Mohamad,

Thank you, I'm so glad you reply :)
Forgive my stupidity, but I don't quite follow. (coming from another IDE and language it's pretty difficult to adapt to flutter/dart)
I have in total 6 players:
```
class _HomeState extends State<Home> {
var player1 = Player.network("https://somewhere.com/play/1.mp3", autoPlay: false);
  var player2 = Player.network("https://somewhere.com/play/2.mp3", autoPlay: false);
    .........etcetera...
then further down below I have:
@override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.black,
      body: FutureBuilder(blablabla..............
................
return SafeArea(
                 child: SingleChildScrollView(
                   child: Column( ................
.................
                       //////////////////////Player//////////////
                       Container( //by using theme dark in Main, no color background is needed
                           width: 400.0,
                           height: 50,
                           child: PlayerBar(player: player1, options: [],
    )
                       ),
```

I'm not sure to where to place this code and how to use that variable subscription.
```
var subscription = player.streams.playing.stream.listen((bool playing) {
   if (playing) {
       player.others.forEach((player) => player.pause());
   }
});
```
Also I thought I should use player.pause(), because the previous players won't work anymore. They should not be disposed, just paused. Or maybe I misinterpret this "and dont forget to close the subscription on dispose method".
I also checked your examples on pub.dev, but I don't quite get it. Sorry.

Thanks for your help and insight.

Best regards,
Jerry

**Reply by mohamadlounnas** [reply]
hi @jeejeestudio
sorry I've been busy lately
u can place it in `initState`
example:
```dart
@override
void initState() {
  var subscription = player.streams.playing.stream.listen((bool playing) {
     if (playing) {
         player.others.forEach((player) => player.pause());
     }
  });
}
```
but I advise you to use a list to make it easy to start and dispose
example:
```dart
var subscriptions = <StreamSubscription>[];
@override
void initState() {
  var players = [player1,player2]; // ...
  players.forEach((player){
    subscriptions.add(player.streams.playing.stream.listen((bool playing) {
       if (playing) {
           player.others.forEach((player) => player.pause());
       }
    }));
  })
}
@override
void dispose() {
  subscriptions.forEach((subscription)=>subscription.cancel())
}
```

**Reply by jeejeestudio** [reply]
Hi @mohamadlounnas
No problem at all. I'm happy you are trying to help.
I will check this out tomorrow to see if I can get it working. Thanks very much!

**Reply by jeejeestudio** [reply]
Hi @mohamadlounnas,

I changed this line as it gave an error:
`subscriptions.add(player.streams.playing.stream.listen((bool playing) {`
to
`subscriptions.add(player.streams.playing.listen((bool playing) {`

and I had to add `import 'dart:async';`

And now it is working as expected :)

Thanks very much!
